edx/frontend-app-discussions
# tests: Clean up and optimise redux data layer and add tests [BD-38] [TNL-8517] [BB-4643]

This change adds tests for the redux data store, moves the normalization of data out of the reducer, and reorganizes data for better access.

**Tickets:** [TNL-8517](https://openedx.atlassian.net/browse/TNL-8517)
**Dependencies**: None

**Screenshots**: NA

**Merge deadline**: "None"

**Testing instructions**:

1. Run the test suite
2. Browse around the app and make sure everything works as expected. 

**Reviewers**
